galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager/src/main/java/dev/galasa/zos3270/spi/Zos3270TerminalImpl.java

```diff
@@ -153,36 +136,51 @@ public Zos3270TerminalImpl(String id, String host, int port, boolean tls, IFrame
 
         getScreen().registerScreenUpdateListener(this);
 
-        storedArtifactsRoot = framework.getResultArchiveStore().getStoredArtifactsRoot();
-        terminalRasDirectory = storedArtifactsRoot.resolve("zos3270").resolve("terminals").resolve(this.terminalId);
+        this.storedArtifactsRoot = framework.getResultArchiveStore().getStoredArtifactsRoot();
+        this.terminalRasDirectory = storedArtifactsRoot.resolve("zos3270").resolve("terminals").resolve(this.terminalId);
+
+        this.liveTerminalUrl = getLiveTerminalUrl(id,host, framework, this.runId , this.terminalId );
+
+        setDeviceTypes(TerminalDeviceTypes.get(image));
+        this.logConsoleTerminals = LogConsoleTerminals.get();
+    }
+
+
+    /** Connect to the zos3270 terminal  */
+    private URL getLiveTerminalUrl(
+        String id, 
+        String host, 
+        IFramework framework, 
+        String runId , 
+        String terminalId
+    ) throws Zos3270ManagerException {
+
+        URL liveTerminalUrl = null ;
+
         URL propLiveTerminalUrl = LiveTerminalUrl.get();
-        if (propLiveTerminalUrl == null) {
-            liveTerminalUrl = null;
-        } else {
+        if (propLiveTerminalUrl != null) {
             try {
                 // *** Register the terminal to the UI which will own the terminal view
                 HttpURLConnection connection = (HttpURLConnection) propLiveTerminalUrl.openConnection();
                 connection.setRequestMethod("HEAD");
-                connection.addRequestProperty("zos3270-runid", this.runId);
-                connection.addRequestProperty("zos3270-terminalid", this.terminalId);
+                connection.addRequestProperty("zos3270-runid", runId);
+                connection.addRequestProperty("zos3270-terminalid", terminalId);
                 connection.setDoInput(true);
                 connection.setDoOutput(false);
                 connection.connect();
```

Comment:
I guess we are assuming that garbage collection of the connection will close the connection, and we don't have to do it explicitly ?